File: SavaariAPI/Location.py
# =========================================================================
# 						 		LOCATION CLASS
# =========================================================================

# Nabeel Danish
class Location(object):

	# Main Constructor
	def __init__(self, *args, **kwargs):

		# Main Attributes
		self.__latitude = kwargs.get('latitude')
		self.__longitude = kwargs.get('longitude')
		self.__timestamp = kwargs.get('timestamp')

		# Super Constructor
		super(Location, self).__init__()

	# End of __init__
	
	# Getters and Setters
	def getLatitude(self):
		return self.__latitude

	def setLatitude(self, latitude):
		self.__latitude = latitude

	def getLongitude(self):
		return self.__longitude

	def setLongitude(self, longitude):
		self.__longitude = longitude

	def getTimstamp(self):
		return self.__timestamp

	def setTimestamp(self, timestamp):
		self.__timestamp = timestamp

# End of Class: Location

File: app/src/main/java/com/example/savaari/ride/entity/Location.java
package com.example.savaari.ride.entity;

import com.google.android.gms.maps.model.LatLng;

public class Location
{
	// Main Attributes
	Double latitude;
	Double longitude;
	Long timestamp;
	
	// Main Constructors
	public Location() {
		super();
	}

	public Location(LatLng latLng) {
		this.latitude = latLng.latitude;
		this.longitude = latLng.longitude;
		this.timestamp = null;
	}
	public Location(Double latitude, Double longitude) {
		super();
		this.latitude = latitude;
		this.longitude = longitude;
		this.timestamp = null;
	}
	public Location(Double latitude, Double longitude, Long timestamp) {
		super();
		this.latitude = latitude;
		this.longitude = longitude;
		this.timestamp = timestamp;
	}
	
	// Getters and Setters
	public Double getLatitude() {
		return latitude;
	}
	public void setLatitude(Double latitude) {
		this.latitude = latitude;
	}
	public Double getLongitude() {
		return longitude;
	}
	public void setLongitude(Double longitude) {
		this.longitude = longitude;
	}
	public Long getTimestamp() {
		return timestamp;
	}
	public void setTimestamp(Long timestamp) {
		this.timestamp = timestamp;
	}
}
